planner/core/exhaust_physical_plans.go

```diff
@@ -1266,6 +1266,12 @@ func (p *baseLogicalPlan) exhaustPhysicalPlans(_ *property.PhysicalProperty) []P
 	panic("baseLogicalPlan.exhaustPhysicalPlans() should never be called.")
 }
 
+func (la *LogicalAggregation) canPushDown() bool {
+	// Child may be of other types in the future, such as limit, topn and projection.
```

Comment:
Please add more comments to explain why we cannot push aggregation down to coprocessor if the child is not DataSource. For example, the check must be called after logical optimization, and current pushed down `Limit` / `TopN` would convert task to rootTask, etc. Also, add a TODO to remind that we should check `Projection` as well if we support `Projection` pushdown later.

Reply:
Okay.